index.bs

```diff
@@ -6081,6 +6101,17 @@ abstract operations are used to implement these "cross-realm transforms".
 
 The following abstract operations are a grab-bag of utilities.
 
+<div algorithm>
+ <dfn abstract-op lt="CanTransferArrayBuffer"
+ id="can-transfer-array-buffer">CanTransferArrayBuffer(|O|)</dfn> performs the following steps:
+
+ 1. Assert: [$Type$](|O|) is Object.
+ 1. Assert: |O| has an \[[ArrayBufferData]] internal slot.
+ 1. If ! [$IsDetachedBuffer$](|O|) is true, return false.
+ 1. If [$SameValue$](|O|.\[[ArrayBufferDetachKey]], undefined) is false, return false.
```

Comment:
This is the important bit. `IsDetachedBuffer(O) === false` does not guarantee that `DetachArrayBuffer(O)` will not throw an error, since the buffer may be "non-detachable" (e.g. if it's a `WebAssembly.Memory`'s buffer). We have to perform the exact same steps as `DetachArrayBuffer(O)`, except for actually detaching the buffer.